utilities/include/vec3.hpp

```diff
@@ -1,120 +1,131 @@
+#ifndef VEC3.HPP
+#define VEC3.HPP
+
 //code by Ameya Vikrama Singh
 #include<iostream>
 #include<cmath>
 class vec3{
+    float x,y,z;
     public:
-        float x,y,z;
-
-        vec3(){
-            x=y=z=0;
-        }
+        vec3() : x{0},y{0},z{0}{}
 
         //Cartesian coordinate Constructor
-        vec3(float x,float y,float z){
-            this->x=x;
-            this->y=y;
-            this->z=z;
-        }
-        
-        //assignment operator
-        void operator=(vec3 s){
-            x=s.x;
-            y=s.y;
-            z=s.z;
-        }
+        vec3(float X,float Y,float Z) : x{X}, y{Y}, z{Z}{}
 
         //norm function
-        float norm(){
+        constexpr float norm() const{
             return sqrt(x*x+y*y+z*z);
         }
 
         //vector addition
-        vec3 operator+(vec3 s){
-            vec3 sum;
-            sum.x=s.x+x;
-            sum.y=s.y+y;
-            sum.z=s.z+z;
-            return sum;
-        }
+        friend vec3 operator+(vec3 const& h, vec3 const & s);
```

Comment:
Don't forget to write the function body. Like we discussed in the meet, two things need to be done here: 1) Change the function signature to have `friend constexpr` instead of just `friend`. 2) Use the `+=` operator overload in the definition of the `+` overload.

This also applies to the other operators (namely, `-`, `*`, `/`).

Reply:
it says constexpr cant have a non-literal return type